src/Containers/Microsoft.NET.Build.Containers/Resources/Strings.resx

```diff
@@ -193,7 +193,7 @@
     <comment>{StrBegin="CONTAINER2015: "}</comment>
   </data>
   <data name="InvalidImageName" xml:space="preserve">
-    <value>CONTAINER2005: The first character of the image name must be a lowercase letter or a digit.</value>
+    <value>CONTAINER2005: The first character of the image name must be a lowercase letter or a digit and all characters in the name must be an alphanumeric character, -, /, or _.</value>
```

Comment:
imo, this error message doesn't resolve the issue well.
- _The first character of the image name must be a lowercase letter or a digit._ is enough reason to fail "----"
- "all characters in the name must be an alphanumeric character, -, /, or _" does not fail "-----"
- if we have differentiate this case, would this wording will be better - "none of provided chars in the image name are supported as the image name" (and have different code possibly)

Reply:
Yeah, I agree - I went with the existing message for implementation speed, but users would be better served with a new message and code

Reply:
I added a new message (with the same code) that we use for the new error case:

```
CONTAINER2005: The inferred image name '{0}' contains entirely invalid characters. The valid characters for an image name are alphanumeric characters, -, /, or _, and the image name must start with an alphanumeric character.
```